### PR TITLE
Add object builtin tester.

### PR DESCRIPTION
**Pull request by sennav**
Much better with the tests passing now :)

### CODE REVIEW

File: docs/content/docs/templates.md [hunk 1]
```diff
@@ -897,6 +897,9 @@ Example:
 #### iterable
 Returns true if the given variable can be iterated over in Tera (ie is an array/tuple).
 
+#### object
+Returns true if the given variable is a hash (ie can be iterated over key, value).
```
Comment (Keats) [inline docs/content/docs/templates.md]: is a hash -> is an object
A hash would be a string :)

Comment (sennav) [reply docs/content/docs/templates.md]: Done

**Comment by Keats** [comment]
Thanks!

**Comment by sennav** [comment]
Thanks, since I got in touch here do you mind to answer a question really quick? I'm trying to use tera for customising output from a CLI tool (maybe this is not the main goal of tera, but anyway) and when I try to print the escape character Tera just escapes the initial '\' into '\\'. I've tried to replace the escape function but that seems to be used only for variables (I'm trying to do a macro that can make text bold for instance). Can I make a PR to allow that? Would you point me in the direction of where it is being escaped (I've looked through the code but I didn't figured it out yet), thanks.

**Comment by Keats** [comment]
The initial what?

**Comment by sennav** [comment]
Oh sorry I didn't checked the preview before sending. I'm trying to create this macro:
```
{%- macro bold(t) -%}
{%- set bold = '\x1b[7m' -%}
{%- set end = '\x1b[0m' -%}
{{ bold }}{{ t }}{{ end }}
{%- endmacro bold -%}
```
and `\x1b[7m` gets transformed into `\\x1b[7m`.

**Pull request merged**